xdsl/transforms/print_to_println.py

```diff
@@ -0,0 +1,180 @@
+from dataclasses import dataclass
+from typing import Iterable, Sequence
+import hashlib
+import re
+
+from xdsl.ir import SSAValue, Attribute, MLContext, Operation
+from xdsl.dialects import print, builtin, arith, llvm
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+    PatternRewriteWalker,
+)
+
+from xdsl.passes import ModulePass
+
+
+i8 = builtin.IntegerType(8)
+
+
+def legalize_str(val: str):
+    """
+    Takes any string and legalizes it to be a global llvm symbol.
+    (for the strictes possible interpreation of this)
+
+     - Replaces all whitespaces and dots with _
+     - Deletes all non ascii alphanumerical characters
+
+    The resulting string consists only of ascii letters, underscores and digits
+    """
+    val = re.sub(r"(\s+|\.)", "_", val)
+    val = re.sub(r"[^A-Za-z0-9_]+", "", val).rstrip("_")
```

Comment:
Would be nice to have a bunch of tests for this and helpers below

Reply:
Tests added!